airlearner/airlearner-xgboost/src/main/scala/com/airbnb/common/ml/xgboost/model/XGBoostModel.scala

```diff
@@ -36,22 +26,67 @@ class XGBoostModel(training: DMatrix, eval: DMatrix)
 
 object XGBoostModel extends ScalaLogging {
 
+  def logLoss(p: Double, y: Double): Double = {
+    -y * math.log(p) - (1 - y) * math.log(1 - p)
+  }
+
+  def linearLoss(p: Double, y: Double): Double = {
+    math.abs((p - y)/y)
+  }
+
+  def eval(training: DMatrix,
+      eval: DMatrix,
+      params: Map[String, Any],
+      lossFunction: (Double, Double) => Double
+  ): Double = {
+    val model = XGBoostModel.train(params, training)
+    val prediction = model.predict(eval)
+
+    val loss = eval.getLabel.zip(prediction).map(
+      a => {
+        math.abs(lossFunction(a._2(0), a._1))
+      }
+    ).sum / prediction.length
+    model.dispose
+    loss
+  }
+
   def getModelByLabeledPoint(
       training: Seq[LabeledPoint],
-      eval: Seq[LabeledPoint]
+      eval: Seq[LabeledPoint],
+      loss: String
   ): XGBoostModel = {
     val trainingDMatrix = new DMatrix(training.iterator, null)
     val evalDMatrix = new DMatrix(eval.iterator, null)
-    new XGBoostModel(trainingDMatrix, evalDMatrix)
+    new XGBoostModel(trainingDMatrix, evalDMatrix, getLossFunction(loss))
   }
 
   def getModelByFile(
       training: String,
-      eval: String
+      eval: String,
+      loss: String
+  ): XGBoostModel = {
+    getModelByFileWithLoss(training, eval, getLossFunction(loss))
+  }
+
+  private def getLossFunction(loss: String): (Double, Double) => Double = {
+    val lossFun: (Double, Double) => Double = loss match {
+      case "linear" => XGBoostModel.linearLoss
+      case _ => XGBoostModel.logLoss
```

Comment:
Maybe push the default up to where it's pulled out of the config, and force this method's param to be a valid name?